### PR TITLE
Add RAG search query generation option

### PR DESCRIPTION
**Pull request by mrmer1**
Two options for RAG search query generation:
- search_queries_only param
- tools

### CODE REVIEW

File: fern/pages/text-generation/retrieval-augmented-generation-rag.mdx [hunk 1]
```diff
@@ -101,6 +103,50 @@ co.chat(
 
 Indeed, to generate a factually accurate answer to the question `"Who is more popular: Nsync or Backstreet Boys?"`, looking up `Nsync popularity` and `Backstreet Boys popularity` first would be helpful.
 
+##### Option 2: Using a tool use approach
```
Comment (maximevoisincohere) [inline fern/pages/text-generation/retrieval-augmented-generation-rag.mdx:106]: maybe "Option 2: Using a Tool"?

File: fern/pages/text-generation/retrieval-augmented-generation-rag.mdx [hunk 2]
```diff
@@ -101,6 +103,50 @@ co.chat(
...
+
+If you are looking for greater control over how search queries are generated, you can use the tool use approach to generate search queries.
```
Comment (maximevoisincohere) [inline fern/pages/text-generation/retrieval-augmented-generation-rag.mdx:108]: you can use Cohere's Tools capabilities to generate search queries

**Review by maximevoisincohere — CHANGES_REQUESTED** [review]
LGTM. Maybe we could add another example that says: you can then customize the tool preamble and/or the tool definition, to generate queries that are more relevant to your use case. 

For example, [[ we do the verbose example from this notebook: https://colab.research.google.com/drive/1oHg6lO-VnXWRFQV5WLLRy6u_SE--jrpG#scrollTo=EeHkg9zet7v5 ]] and show that it generates more queries than before?

**Review by mrmer1 — COMMENTED** [review]
done

**Review by billytrend-cohere — APPROVED** [review]

**Review by trentfowlercohere — APPROVED** [review]
Looks good to me!

**Review by lfayoux — APPROVED** [review]

**Pull request merged**